main.go

```diff
@@ -876,18 +861,86 @@ func (p *gateway) runningLoop(ctx context.Context) (err error) {
 	}
 }
 
+// getSchedulerReportingDelay evaluates config to determine the reporting delay time for schedulers
+func getSchedulerReportingDelay(config *config.GatewayConfig) int64 {
+	var reportingDelayNs int64
+
+	// only set ReportingDelay on the scheduler if the internal metrics server is off
+	// or the local debug server is on.
+	if (config.LocalDebugServer != nil && *config.LocalDebugServer != "") || (config.InternalMetricsListenerAddress == nil || *config.InternalMetricsListenerAddress == "") {
```

Comment:
i'm confused by "local debug server" here making a reporting delay ?

Reply:
So the issue is that the debug server relies on `scheduler.Var()` which returns a datapoint cache in the scheduler. The scheduler populates that cache on the scheduled reporting interval.  So, in order to get the payload for the debug server we have to have a scheduled collection interval.

If the internal metircs listener is on we don't need to schedule datapoint collection because the call to CollectDatapoints() is on demand and does not use the datapoint\ cache.  The exception to that rule is when the debug server is on, because we need to populate that datapoint cache.

Instead of always scheduling collection, this optimizes the case where the internal metrics server is on and the the debug server is off.  This is the only case where scheduling collection can be left off.

Reply:
I would remove the connection between the debug server and the scheduler and just make it call the same code the scraper calls.

Reply:
ok cool, so I linked the debug server to the scheduler's `CollectDatapoints()` method. So it returns current values on demand.